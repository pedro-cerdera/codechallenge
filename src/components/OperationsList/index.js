import OperationsList from './OperationsList';

export {OperationsList};
