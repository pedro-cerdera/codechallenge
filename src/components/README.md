
# Components

Here we have some components

- [OperationsList](/src/components/OperationsList/README.md)
- [Account](/src/components/Account/README.md)
- [Input](/src/components/Input/README.md)
- [TransactionForm](/src/components/forms/TransactionForm/README.md)
