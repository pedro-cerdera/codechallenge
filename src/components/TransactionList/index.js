import TransactionList from './TransacitonList';

export {TransactionList};
