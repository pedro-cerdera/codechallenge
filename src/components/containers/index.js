export * from './SafeAreaContainer';
export * from './CardContainer';
export * from './DefaultScapingContainer';
