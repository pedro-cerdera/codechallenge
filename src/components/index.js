export * from './containers';
export * from './forms';
export * from './Account';
export * from './Icon';
export * from './OperationsList';
export * from './Input';
