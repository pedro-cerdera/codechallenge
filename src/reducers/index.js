import {TransactionReducer, Actions as TranscationActions} from './transaction';

export default {TransactionReducer};

export const Actions = {Transaction: TranscationActions};
