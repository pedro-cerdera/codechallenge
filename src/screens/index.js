export * from "./Home";
export * from "./NewTransaction";
export * from "./Transactions";