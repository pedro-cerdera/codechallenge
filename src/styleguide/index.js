import colors from './Colors';
import typografy from './Typografy';

export default {colors, typografy};
