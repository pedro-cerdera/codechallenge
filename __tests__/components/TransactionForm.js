import React from 'react';
import {render, fireEvent, cleanup} from 'react-native-testing-library';

import {TransactionForm} from '../../src/components';

const mockSubmit = jest.fn();

jest.mock('../../src/components/Icon/Icon', () => 'Icon');

describe('Testing Transaction Form', () => {
  beforeEach(() => cleanup());

  it('make a incorrecly transcation', () => {
    const {getByTestId} = render(<TransactionForm onSubmit={mockSubmit} />);
    fireEvent.press(getByTestId('button'));

    expect(mockSubmit).not.toHaveBeenCalled();
  });

  it('make a correcly transaction', () => {
    const {getByTestId} = render(<TransactionForm onSubmit={mockSubmit} />);
    fireEvent.changeText(getByTestId('value'), '20');
    fireEvent.changeText(getByTestId('description'), 'Test');
    fireEvent.press(getByTestId('button'));

    expect(mockSubmit).toHaveBeenCalledTimes(1);
    expect(mockSubmit).toHaveBeenCalledWith(-20, 'Test');
  });

  it('renders correctly', () => {
    const tree = render(<TransactionForm onSubmit={mockSubmit} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
