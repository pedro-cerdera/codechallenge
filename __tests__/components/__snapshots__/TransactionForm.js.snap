// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Transaction Form renders correctly 1`] = `
<View
  onLayout={[Function]}
  style={
    Array [
      Object {
        "flex": 1,
        "maxHeight": 350,
        "width": "100%",
      },
      Object {
        "paddingBottom": 0,
      },
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "white",
          "borderRadius": 5,
          "elevation": 1,
          "maxWidth": 500,
          "shadowColor": "gray",
          "shadowOffset": Object {
            "height": 3,
            "width": 0,
          },
          "shadowOpacity": 1,
          "shadowRadius": 3,
          "width": "100%",
        },
        Object {
          "flex": 1,
          "justifyContent": "space-between",
          "padding": 20,
        },
      ]
    }
  >
    <RCTScrollView
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <View>
        <View
          style={
            Object {
              "display": "flex",
              "marginTop": 0,
            }
          }
        >
          <Text>
            Valor
          </Text>
          <TextInput
            allowFontScaling={true}
            keyboardType="numeric"
            masked={true}
            onBlur={[Function]}
            onChangeText={[Function]}
            placeholder="R$ 100,00"
            placeholderTextColor="#999999"
            rejectResponderTermination={true}
            style={
              Array [
                Object {
                  "backgroundColor": "white",
                  "borderColor": "black",
                  "borderRadius": 25,
                  "borderStyle": "solid",
                  "borderWidth": 1,
                  "height": 50,
                  "justifyContent": "center",
                  "marginBottom": 0,
                  "marginLeft": 0,
                  "marginRight": 0,
                  "marginTop": 0,
                  "maxHeight": 200,
                  "paddingBottom": 15,
                  "paddingLeft": 19,
                  "paddingRight": 19,
                  "paddingTop": 15,
                },
              ]
            }
            testID="value"
            title="Valor"
            topSpacing={false}
            type="money"
            underlineColorAndroid="transparent"
            value="R$0,00"
            visible={true}
          />
        </View>
        <View
          style={
            Object {
              "display": "flex",
              "marginTop": 15,
            }
          }
        >
          <Text>
            Descrição
          </Text>
          <TextInput
            allowFontScaling={true}
            keyboardType="default"
            masked={false}
            multiline={true}
            numberOfLines={2}
            onBlur={[Function]}
            onChangeText={[Function]}
            placeholder="Ex.: Transferência bancária"
            placeholderTextColor="#999999"
            rejectResponderTermination={true}
            style={
              Array [
                Object {
                  "backgroundColor": "white",
                  "borderColor": "black",
                  "borderRadius": 25,
                  "borderStyle": "solid",
                  "borderWidth": 1,
                  "justifyContent": "center",
                  "marginBottom": 0,
                  "marginLeft": 0,
                  "marginRight": 0,
                  "marginTop": 0,
                  "maxHeight": 200,
                  "minHeight": 50,
                  "paddingBottom": 15,
                  "paddingLeft": 19,
                  "paddingRight": 19,
                  "paddingTop": 15,
                },
              ]
            }
            testID="description"
            title="Descrição"
            topSpacing={true}
            underlineColorAndroid="transparent"
            value=""
            visible={true}
          />
        </View>
        <View
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "justifyContent": "center",
              "marginTop": 15,
            }
          }
        >
          <Text>
            Saque
          </Text>
          <RCTSwitch
            accessibilityRole="switch"
            onChange={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "height": 31,
                "width": 51,
              }
            }
            testID="switch"
            value={false}
          />
          <Text>
            Depósito
          </Text>
        </View>
      </View>
    </RCTScrollView>
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#1C2124",
          "borderRadius": 25,
          "opacity": 1,
          "padding": 15,
          "width": "100%",
        }
      }
      testID="button"
    >
      <Text
        style={
          Object {
            "color": "white",
          }
        }
      >
        Fazer Transação
      </Text>
    </View>
  </View>
</View>
`;
